src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -281,8 +286,27 @@ public SeekableStream make() {
 
 
     PathInputResource(final Path pathResource) {
+        this(pathResource, Function.identity());
+    }
+
+    PathInputResource(final Path pathResource, Function<SeekableByteChannel, SeekableByteChannel> wrapper) {
         super(Type.PATH);
         this.pathResource = pathResource;
+        this.wrapper = wrapper;
+    }
+
+    /** Returns a modified PathInputResource with the specific wrapper. After calling this,
+     * do NOT use the original object anymore.
+     *
+     * @param wrapper
+     * @return wrapped PathInputResource
+     */
+    PathInputResource wrap(Function<SeekableByteChannel, SeekableByteChannel> wrapper) {
```

Comment:
If we need it for gatk then we can keep it.  It's not a public method, so I was assuming no one is using it if they're not in htsjdk.  Can you check if your gatk branch needs it for something?